### DISCUSSION TITLE
Schedule a workflow programatically

### DISCUSSION THREAD

**Discussion by cwgw** [community/community#25960] [Actions]
<p>Is it possible to schedule a workflow with a variable or dynamic value? I’d like to be able to set the cron for a workflow via a rest api call.</p>
<p>I imagine using something like the following:</p>
<pre><code class="lang-yaml">on:
  schedule:
    - cron: ${{ github.event.inputs.cron }}
</code></pre>
<p>But of course there’s no event to provide inputs in this example. I thought maybe I could use <code>env</code> context instead, but according to the docs:</p>
<blockquote>
<p>You can only use the <code>env</code> context in the value of the <code>with</code> and <code>name</code> keys, or in a step’s <code>if</code> conditional.</p>
</blockquote>
<p>If you have any thoughts on setting a scheduled workflow via a rest api call, I’d love to hear them.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/cwgw">@cwgw</a>,</p>
<p>Glad to see you in Github Community Forum!</p>
<p>Dynamic schedule setting is not supported, you cannot dynamically set the value.</p>
<p>As an alternative, you can try to update the schedule setting and push to repository.<br>
Two options here:</p>
<ol>
<li>Directly update cronsetting via git command.</li>
<li>If you’d like to use rest api, you can use <a href="https://developer.github.com/v3/repos/#create-a-repository-dispatch-event" rel="nofollow noopener">repository_dispatch</a> event, add the cornsetting in the payload value, trigger repository_dispatch event to update the cornsetting.<br>
My workflow shows the cornsetting value is updated.<br>
<a href="https://github.com/weide-zhou/schedulewf1/runs/868756211?check_suite_focus=true#step:5:9" rel="nofollow noopener">https://github.com/weide-zhou/schedulewf1/runs/868756211?check_suite_focus=true#step:5:9</a><br>
<img src="https://user-images.githubusercontent.com/52772761/181094309-35b735dd-5d13-43a1-8e9c-8561a7901d3e.png" alt="image" data-base62-sha1="tGhjBgH54H5xWefPfLLLyWAu7yJ" width="666" height="368">
</li>
</ol>
<p>Thanks</p>

**Reply by mariusfilipowski** [reply]
I've just stumbled upon this solution. But now the example link to the weide-zhou repository is not valid anymore. Is there any updated example here?

**Comment by cwgw** [comment]
<p>Thank you for your response <a class="mention" href="/weide-zhou">@weide-zhou</a>. I’ll consider implementing your solution in my project.</p>
<p>Cheers.</p>

**Comment by hyprstack** [comment]
Five years on since last comment and this still isn't dynamic? I'm trying to set the cron value using Github vars as I don't want to have to repeat the cron expression throughout my file in conditionals for other steps and would rather reference it either using the vars or env property so that I only ever have to change it in one place, like:

```
name: example_job

on:
  schedule:
    - cron: ${{ vars.DOCKER_BUILD_CRON }}
```

OR

```
name: example_job

env:
  DOCKER_BUILD_CRON: "0 0 * * 1"

on:
  schedule:
    - cron: ${{ env.DOCKER_BUILD_CRON }}
```

With vars I'm getting the following Invalid Workflow File error message: Unrecognized named-value: 'vars'. Located at position 1 within expression: vars.DOCKER_BUILD_CRON

Why is this still not possible and what's the workaround without repeating the cron value everywhere?
